From patchwork Fri Jan 12 05:11:14 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: ui/ncurses: Clear remaining space when drawing help line
X-Patchwork-Submitter: Samuel Mendoza-Jonas <sam@mendozajonas.com>
X-Patchwork-Id: 859509
Message-Id: <20180112051114.22519-1-sam@mendozajonas.com>
To: petitboot@lists.ozlabs.org
Cc: Samuel Mendoza-Jonas <sam@mendozajonas.com>
Date: Fri, 12 Jan 2018 16:11:14 +1100
From: Samuel Mendoza-Jonas <sam@mendozajonas.com>
List-Id: Petitboot bootloader development <petitboot.lists.ozlabs.org>

When drawing the screen's help line clear each character after the new
help line to avoid stale parts of the previous screen's help line
remaining on the screen.

Signed-off-by: Samuel Mendoza-Jonas <sam@mendozajonas.com>
---
 ui/ncurses/nc-scr.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ui/ncurses/nc-scr.c b/ui/ncurses/nc-scr.c
index a02627b..0200d01 100644
--- a/ui/ncurses/nc-scr.c
+++ b/ui/ncurses/nc-scr.c
@@ -77,6 +77,10 @@ void nc_scr_frame_draw(struct nc_scr *scr)
 
 	mvwhline(scr->main_ncw, LINES - nc_scr_pos_help_sep, 1, ACS_HLINE,
 		COLS - 2);
+	if (1 + strlen(scr->frame.help) < COLS)
+		mvwhline(scr->main_ncw, LINES - nc_scr_pos_help,
+				1 + strlen(scr->frame.help), ' ',
+				COLS - 1 - strlen(scr->frame.help));
 	mvwaddstr(scr->main_ncw, LINES - nc_scr_pos_help, 1, scr->frame.help);
 	nc_scr_status_draw(scr);
 }
